pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -212,54 +317,57 @@ func (r *Runtime) RunGadget(gadgetCtx runtime.GadgetContext) (runtime.CombinedGa
 	results := make(runtime.CombinedGadgetResult)
 	var resultsLock sync.Mutex
 
-	allParams := make(map[string]string)
-	gadgets.ParamsToMap(
-		allParams,
-		gadgetCtx.GadgetParams(),
-		gadgetCtx.RuntimeParams(),
-		gadgetCtx.OperatorsParamCollection(),
-	)
-
-	gadgetCtx.Logger().Debugf("Params")
-	for k, v := range allParams {
-		gadgetCtx.Logger().Debugf("- %s: %q", k, v)
-	}
-
 	wg := sync.WaitGroup{}
-	for _, pod := range pods {
+	for _, t := range targets {
 		wg.Add(1)
-		go func(pod gadgetPod) {
+		go func(pod target) {
 			gadgetCtx.Logger().Debugf("running gadget on node %q", pod.node)
-			res, err := r.runGadget(gadgetCtx, pod, allParams)
+			res, err := r.runGadget(gadgetCtx, pod, paramMap)
 			resultsLock.Lock()
 			results[pod.node] = &runtime.GadgetResult{
 				Payload: res,
 				Error:   err,
 			}
 			resultsLock.Unlock()
 			wg.Done()
-		}(pod)
+		}(t)
 	}
 
 	wg.Wait()
 	return results, results.Err()
 }
 
-func (r *Runtime) runGadget(gadgetCtx runtime.GadgetContext, pod gadgetPod, allParams map[string]string) ([]byte, error) {
+func (r *Runtime) dialContext(dialCtx context.Context, target target) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	}
+
+	// If we're in Kubernetes connection mode, we need a custom dialer
+	if r.connectionMode == ConnectionModeKubernetesProxy {
+		opts = append(opts, grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+			return NewK8SExecConn(ctx, target, time.Second*ConnectTimeout)
+		}))
+	}
+
+	conn, err := grpc.DialContext(dialCtx, "passthrough:///"+target.name, opts...)
```

Comment:
What's `passthrough:///` for here. I tried to look around in the grpc and I can't find documentation about it.

Reply:
It's a scheme handler defined [here](https://pkg.go.dev/google.golang.org/grpc/internal/resolver/passthrough), the resolution of targets is partially documented [here](https://github.com/grpc/grpc/blob/master/doc/naming.md). Before, we only used a custom dialer (to tunnel through the Kubernetes API server) and left that target string empty, but in direct mode this is actually used to hand over the target. I'll need to re-check why I couldn't use `unix://...` or `tcp://` directly, though (too long ago).

Reply:
If this is still required, could you please add a comment in the code with the your github comment above?